design_documents/transactional_testing/transactional_testing.md

```diff
@@ -0,0 +1,75 @@
+# Transactional testing proposal
+
+## Context
+
+Some flows are not currently intuitively testable via standard testing methods, 
+like accounts logic (__validate__, __execute__ etc.), transaction rejections,
+fees, contract upgrade flows, and other e2e testing procedures.
+
+## Goal
+
+In order to facilitate that, we should consider adding a way to test
+the contracts in conditions as close to real networks conditions as possible.
+
+Also, the goal is to make the deployments testable and validation 
+of the deployments possible to the user. 
+
+The tests should be also runnable in a variety of environments, depending on the
+use case (namely katana, devnet, or testnet). This would increase user's confidence
+in the release process.
+
+## Proposed solution
+
+We could consider extending `cast script`, since the functionalities would overlap.
+This could be considered as a variant of `cast script`, in a way. 
```

Comment:
Mix of forge test and cast script

Reply:
I wonder if this should be a part of forge test or we should have a specilized tool for it 🤔

Reply:
I think this desing doc should decide on something

Reply:
I think it already does, since the alternative approaches are integrating it into either one of the existing ones.
I am proposing a new tool for that.

Reply:
@THenry14 idea is to merge cast_std and forge_std, just provide different backends for them. Isn't it a good idea to do that first and then implement deployment scripts as another backend?